edm4eic.yaml

```diff
@@ -433,22 +433,16 @@ datatypes:
 
   edm4eic::Vertex:
     Description: "EIC vertex"
-    Author: "W. Armstrong, S. Joosten, based off EDM4hep"
+    Author: "J. Osborn"
     Members:
-      - int32_t             primary       // Boolean flag, if vertex is the primary vertex of the event
+      - int32_t             type       // Type flag, to identify what type of vertex it is (e.g. primary, secondary, generated, etc.)
```

Comment:
Can we define the types through getters in `ExtraCode`? See e.g. https://github.com/key4hep/EDM4hep/blob/main/edm4hep.yaml#L188-L220 and the respective `MutableExtraCode` above it.

Reply:
This probably makes sense but if it is okay with you I'd rather defer this to another PR. I don't think we have discussed all of the different types we need to be able to support. This requires input from S&C, background, tracking, and probably others (HF working group...). I can think of a starting list but it is not immediately clear to me that this will be exhaustive.

Reply:
Ok, sounds good to keep out of scope here. Can you create an issue about that, post it here, and resolve the conversation?

Reply:
Done https://github.com/eic/EDM4eic/issues/62